src/Stateless/StateMachine.cs

```diff
@@ -600,5 +604,18 @@ public void OnTransitioned(Action<Transition> onTransitionAction)
             if (onTransitionAction == null) throw new ArgumentNullException(nameof(onTransitionAction));
             _onTransitionedEvent.Register(onTransitionAction);
         }
+
+        /// <summary>
+        /// Registers a callback that will be invoked every time the statemachine
+        /// has an internal transition.
+        /// </summary>
+        /// <param name="onInternalTransitionAction">
+        /// The action to execute, accepting the details of the transition
+        /// </param>
+        public void OnTransitionedInternal(Action<Transition> onInternalTransitionAction)
```

Comment:
if this is now a public method, I would be inclined to drop the `internal` keyword
```suggestion
        public void OnTransitioned(Action<Transition> onTransitionAction)
```